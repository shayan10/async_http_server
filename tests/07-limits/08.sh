#!/bin/bash
# echo limits and post limits

PORT=$@

EOL=$'\r\n'

FIRST=$'this should not be changed'
REQUEST=$'POST /write HTTP/1.1'${EOL}$'Content-Length: '${#FIRST}${EOL}${EOL}${FIRST}
printf "$REQUEST" | nc -N 127.0.0.1 $PORT

DATA=`printf "%1300s" | tr " " "N"`
EREQUEST=$'GET /echo HTTP/1.1'${EOL}$'Header1: '${DATA}${EOL}${EOL}
PREQUEST=$'POST /write HTTP/1.1'${EOL}$'Content-Length: '${#DATA}${EOL}${EOL}${DATA}
RREQUEST=$'GET /read HTTP/1.1'${EOL}${EOL}

for _ in $(seq 1 2); do
    printf "$EREQUEST" | nc -N 127.0.0.1 $PORT
    printf "\n"
    printf "$PREQUEST" | nc -N 127.0.0.1 $PORT
    printf "\n"
    printf "$RREQUEST" | nc -N 127.0.0.1 $PORT
    printf "\n"
done
