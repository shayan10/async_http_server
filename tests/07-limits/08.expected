HTTP/1.1 200 OK
Content-Length: 26

this should not be changedHTTP/1.1 413 Request Entity Too Large


HTTP/1.1 413 Request Entity Too Large


HTTP/1.1 200 OK
Content-Length: 26

this should not be changed
HTTP/1.1 413 Request Entity Too Large


HTTP/1.1 413 Request Entity Too Large


HTTP/1.1 200 OK
Content-Length: 26

this should not be changed
